File: python/GameGridEx/Ex04.py
# Ex04.py

from ch.aplu.jgamegrid import GameGrid, Actor, Location
from ch.aplu.util import X11Color


# --------------------- class Bear ---------------------
class Bear(Actor):
   def __init__(self):
      Actor.__init__(self, "sprites/bear.gif");
  
   def act(self):
      if self.isMoveValid():
         self.move()
      else:
         if self.isHorzMirror():
            self.setHorzMirror(False)
            self.turn(270)
            self.move()
            self.turn(270)
         else:
            self.setHorzMirror(True)
            self.turn(90)
            self.move()
            self.turn(90)
      self.tryToEat()
      
   def tryToEat(self):
      actor = gg.getOneActorAt(self.getLocation(), Leaf)
      if actor != None:
         actor.hide()
    
# --------------------- class Leaf ---------------------------
class Leaf(Actor):
   def __init__(self):
      Actor.__init__(self, "sprites/leaf.gif")

# ----------------- main -----------------------------
gg = GameGrid(10, 10, 60, X11Color("red"))
gg.addActor(Bear(), Location(0, 0))
for i in range(10):
   gg.addActor(Leaf(), gg.getRandomEmptyLocation())
gg.show()

File: tutorial/Ex04.java
// Ex04.java
// Cell collisions

import ch.aplu.jgamegrid.*;

public class Ex04 extends GameGrid
{
  public Ex04()
  {
    super(10, 10, 50, java.awt.Color.red);
    addActor(new Bear(Leaf.class), new Location(0, 0));
    for (int i = 0; i < 10; i++)
      addActor(new Leaf(), getRandomEmptyLocation());
    show();
  }

  public static void main(String[] args)
  {
    new Ex04();
  }
}
